src/page/hugepage.rs

```diff
@@ -338,12 +338,12 @@ fn priv_hugepage_supported(exp: usize) -> bool {
 
     use self::libc::{stat, lstat, ENOENT, ENOMEM, c_char};
     use self::errno::errno;
-    use core::mem::uninitialized;
+    use core::mem::MaybeUninit;
 
     let path = get_linux_hugepage_directory!(exp);
 
-    let mut s = unsafe { uninitialized::<stat>() };
-    if unsafe { lstat(path.as_ptr() as *const c_char, &mut s) } < 0 {
+    let s: MaybeUninit<stat> = unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
I'm not super familiar with this new api but can you explain why assume_init is making a `MaybeUninit` here? In testing the following work:

```rust
    let mut s: stat = unsafe { MaybeUninit::uninit().assume_init() };
    unsafe {
        let x = lstat(c.as_ptr() as *const c_char, &mut s);
        println!("{}", x);
    }
```

and

```rust
    let s: MaybeUninit<stat> = MaybeUninit::uninit();
    unsafe {
        let x = lstat(c.as_ptr() as *const c_char, &mut s.assume_init());
        println!("{}", x);
    }
```

so I'm a bit confused why assume_init is being used in both places.